skops/io/_sklearn.py

```diff
@@ -240,17 +248,25 @@ def _construct(self):
 
 # tuples of type and function that gets the state of that type
 GET_STATE_DISPATCH_FUNCTIONS = [
-    (LossFunction, sgd_loss_get_state),
     (Tree, tree_get_state),
 ]
+if not SKLEARN_GT_15:
+    GET_STATE_DISPATCH_FUNCTIONS.append((LossFunction, sgd_loss_get_state))
+
 for type_ in UNSUPPORTED_TYPES:
     GET_STATE_DISPATCH_FUNCTIONS.append((type_, unsupported_get_state))
 
 # tuples of type and function that creates the instance of that type
-NODE_TYPE_MAPPING = {
-    ("SGDNode", PROTOCOL): SGDNode,
-    ("TreeNode", PROTOCOL): TreeNode,
-}
+if not SKLEARN_GT_15:
+    NODE_TYPE_MAPPING = {
+        ("TreeNode", PROTOCOL): TreeNode,
+        ("SGDNode", PROTOCOL): SGDNode,
+    }
+else:
+    NODE_TYPE_MAPPING = {
+        ("TreeNode", PROTOCOL): TreeNode,
+    }
```

Comment:
```suggestion
NODE_TYPE_MAPPING = {
    ("TreeNode", PROTOCOL): TreeNode,
}
if SKLEARN_SGD_LOSS:
    NODE_TYPE_MAPPING.update({
        ("SGDNode", PROTOCOL): SGDNode,
    })
```

Reply:
I remember now why I wrote it weird like that without the `update`, it was because of this persistent error otherwise: 

`skops/io/_sklearn.py:266: error: Dict entry 0 has incompatible type "tuple[str, int]": "type[SGDNode]"; expected "tuple[str, int]": "type[TreeNode]"  [dict-item]`

@adrinjalali any idea how to go around it otherwise?

Reply:
I temporarily reverted it so I can work cleaner with the other errors in https://github.com/skops-dev/skops/pull/447/commits/d8da963c808c06efd3204c61afdd45710209280c, if there is a better solution I can change it.

Reply:
You can add explicit typehints when you define the dict, that would solve the issue.

Reply:
thanks addressed in https://github.com/skops-dev/skops/pull/447/commits/c367109de46b1ab289f34f94f9c11d97dd9a16d2